nflow-engine/src/test/java/io/nflow/engine/NflowEngineTest.java

```diff
@@ -0,0 +1,121 @@
+package io.nflow.engine;
+
+import io.nflow.engine.config.db.H2DatabaseConfiguration;
+import io.nflow.engine.service.DummyTestWorkflow;
+import io.nflow.engine.service.WorkflowInstanceInclude;
+import io.nflow.engine.workflow.definition.AbstractWorkflowDefinition;
+import io.nflow.engine.workflow.definition.WorkflowState;
+import io.nflow.engine.workflow.executor.WorkflowExecutor;
+import io.nflow.engine.workflow.instance.QueryWorkflowInstances;
+import io.nflow.engine.workflow.instance.WorkflowInstance;
+import org.joda.time.DateTime;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+import org.springframework.jdbc.datasource.init.ResourceDatabasePopulator;
+
+import javax.sql.DataSource;
+import java.util.*;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Arrays.asList;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.springframework.jdbc.datasource.init.DatabasePopulatorUtils.execute;
+
+public class NflowEngineTest {
+    private static final Logger logger = LoggerFactory.getLogger(NflowEngineTest.class);
+
+    /**
+     * Travis ci build sets env variable SPRING_PROFILES_ACTIVE=nflow.db.$DB
+     * This will enable scanning for io.nflow.engine.config.db which will create
+     * an extra data source, from e.g. H2DatabaseConfiguration.java
+     * The spring setup will fail because there are two competing dataSources.
+     *
+     * This uses system property `spring.profiles.active`
+     * to override env variable `SPRING_PROFILES_ACTIVE`.
+     */
+    @BeforeEach
+    public void setup() {
+        System.setProperty("spring.profiles.active", "nflow.db.dummy");
```

Comment:
What if you would register the wanted DataSource as Primary?
Maybe using `https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/config/BeanDefinitionCustomizer.html` as extra argument in the `registerBean`

Reply:
If I do that the end result seems to be that Spring finds 0 suitable dataSource beans and fails.